pandas/io/json/_json.py

```diff
@@ -1095,6 +1097,8 @@ def _parse_no_numpy(self):
                 for k, v in loads(json, precise_float=self.precise_float).items()
             }
             self.check_keys_split(decoded)
+            if "index" in decoded:
+                decoded["index"] = np.transpose(decoded["index"]).tolist()
```

Comment:
so this could come back as a 2d
list?

Reply:
Hmm rather than this being a JSON issue I wonder if there are more general problems with the index argument in constructors. For instance:

```python
>>> pd.DataFrame([[1], [2]], index=[["a", "b"], ["c", "d"]])
     0
a c  1
b d  2
```

seems to generate the wrong index, so if we fix there might not even need this

Reply:
@jreback If the decoded index corresponds to a MultiIndex one, it will output 2D list.
If it is a single Index, we got an unchanged 1D list.

Reply:
@WillAyd I considered it and thought there may be many features using Index constructor in this way, so it would break a lot.
I will test it.

Reply:
Considering modifications in both DataFrame/Series constructors instead of during json decoding, a few tests would need to be modified (typically transposing index within these tests).
I can rename this PR and work on it.

Reply:
Can you share one of the failing tests? Might need to discuss whether this is a bug fix or an API breaking change first

Reply:
For example :
- pandas/tests/indexing/multiindex/test_loc.py::TestMultiIndexLoc::test_loc_multiindex_list_missing_label 
-> index must be transposed within test.

Reply:
Hmm yea I'm not sure it really matters in that test though. @jreback do you have an opinion here? I think the index argument for a DataFrame constructor has this backwards but maybe there's a historical context I'm unaware of

Reply:
Some other tests fail because the transposing is not consistent enough. Transposing directly in Series/DataFrame constructors should handle numpy arrays, list of numpy arrays, mixed types...
But it should not be a big issue.

Reply:
Do you have any news on this topic ?